euclidigons.lua

```diff
@@ -532,6 +541,113 @@ function redraw()
 		end
 	end
 	screen.update()
+
+	if edit_shape ~= nil then
+		-- ARC 1: Note
+		local scale_degrees = #scale
+		local scale_mark = 64 / scale_degrees
+		local degree = (edit_shape.edits.note - 1) % scale_degrees
+		degree = math.ceil(scale_mark * degree) + 1
+		for i = 1, 64 do
+			if math.floor((i - 1) % scale_mark) == 0 then
+				a:led(1, i, 3)
+			else
+				a:led(1, i, 0)
+			end
+		end
+		a:led(1, degree - 2, 3)
+		a:led(1, degree - 1, 8)
+		a:led(1, degree, 15)
+		a:led(1, degree + 1, 8)
+		a:led(1, degree + 2, 3)
+		-- ARC 2: Octave
+		local octave = util.clamp(6 + math.floor((edit_shape.edits.note - 1) / #scale), 1, 11)
+		for i = 1, 64 do
+			if (i - 3) % 6 == 0 then
+				a:led(2, i, 3)
+			else
+				a:led(2, i, 0)
+			end
+		end
+		a:led(2, octave * 6 - 5, 3)
+		a:led(2, octave * 6 - 4, 8)
+		a:led(2, octave * 6 - 3, 15)
+		a:led(2, octave * 6 - 2, 8)
+		a:led(2, octave * 6 - 1, 3)
+		-- ARC 3: Speed
+		local theta = 64 * edit_shape.theta / tau 
+		while theta < 0 do
+			theta = theta + 64
+		end
+		local theta_i = math.ceil(theta)
+		local brightness = math.floor((theta_i - theta) * 15)
+		for i = 1, 64 do
+			a:led(3, i, 0)
+		end
+		a:led(3, theta_i, brightness)
+		a:led(3, math.fmod(theta_i + 1, 64), 15 - brightness)
+		-- ARC 4: Number of sides
+		for i = 1, 9 do
+			if edit_shape.n >= i then
+				a:led(4, i * 7 - 6, 15)
+				a:led(4, i * 7 - 5, 15)
+				a:led(4, i * 7 - 4, 8)
+				a:led(4, i * 7 - 3, 7)
+				a:led(4, i * 7 - 2, 6)
+				a:led(4, i * 7 - 1, 5)
+				a:led(4, i * 7, 4)
+			else
+				a:led(4, i * 7 - 6, 0)
+				a:led(4, i * 7 - 5, 0)
+				a:led(4, i * 7 - 4, 0)
+				a:led(4, i * 7 - 3, 0)
+				a:led(4, i * 7 - 2, 0)
+				a:led(4, i * 7 - 1, 0)
+				a:led(4, i * 7, 0)
+			end
+		end
+		if edit_shape.n == 9 then
+			a:led(4, 64, 2)
+		else
+			a:led(4, 64, 0)
+		end
+	end
+	a:refresh()
+end
+
+function a.delta(n, d)
```

Comment:
This block is handling of Arc encoder changes. Arc encoders are very precise so note here that we need `delta_arc_*` to store the raw updates and only actually change values when we cross some threshold (that's in `Shape:tick()`).

Also, when we change direction, we want to reset the already stored values since otherwise you'd have to travel twice the distance in this case to have the change register.